src/main/java/org/opensearch/security/ssl/transport/SecuritySSLRequestHandler.java

```diff
@@ -113,9 +113,9 @@ public final void messageReceived(T request, TransportChannel channel, Task task
 
             if (channel instanceof TaskTransportChannel) {
                 final TransportChannel inner = ((TaskTransportChannel) channel).getChannel();
-                nettyChannel = (Netty4TcpChannel) ((TcpTransportChannel) inner).getChannel();
-            } else if (channel instanceof TcpTransportChannel) {
-                final TcpChannel inner = ((TcpTransportChannel) channel).getChannel();
+                nettyChannel = (Netty4TcpChannel) ((BaseTcpTransportChannel) inner).getChannel();
+            } else if (channel instanceof BaseTcpTransportChannel) {
+                final TcpChannel inner = ((BaseTcpTransportChannel) channel).getChannel();
```

Comment:
Lets add a test(s) that confirms this fix is resolved.

If you are looking for an example of how these tests could be written, take a look at https://github.com/opensearch-project/security/blob/main/src/integrationTest/java/org/opensearch/security/http/OnBehalfOfJwtAuthenticationTest.java that might provide some guidance on cluster configuration and a basic setup.

Reply:
@peternied, I have written a Integ test case and its failing for 2 more issues. Where RestRequest::getHttpChannel is being type casted to Netty4HttpChannel. At one place this is already been reported by @cwperks and in addition to that one check is here https://github.com/opensearch-project/security/blob/e69013651189238f3c93d57415d37134c8bfd6df/src/main/java/org/opensearch/security/filter/OpenSearchRequest.java#L47

Reply:
@Gaganjuneja @peternied I see at least 2 issues with this code:
 - it very coupled with Netty, that should not be the case
 - the issue will come back when we would instrument `TaskTransportChannel`

Would I would like to suggest: add `get` (and possibly `put`, have to look into it) methods to `HttpChannel` and `TcpChannel`:

```
public T get(String attribute, Class<T> clazz);
```

With ytat, we could remove `|| !(underlyingRequest.getHttpChannel() instanceof Netty4HttpChannel)) {` from `OpenSearchRequest` since it will be `null` value anyway.

Reply:
Thanks @reta, Yes it would be more cleaner approach.

Reply:
There is one more change where we are trying to get the SSLHandler from the inbound and outbound pipeline

https://github.com/opensearch-project/security/blob/e69013651189238f3c93d57415d37134c8bfd6df/src/main/java/org/opensearch/security/filter/OpenSearchRequest.java#L54

Reply:
Ir does not matter, it is all covered, will send the pull request shortly

Reply:
https://github.com/opensearch-project/OpenSearch/pull/10562

Reply:
One clarification, Please pardon me for the naive question. How will we access the channel attributes (NettyAttribute::proFrom) for authentication? Or the old implementation should be suffice.

Reply:
> One clarification, Please pardon me for the naive question. How will we access the channel attributes (NettyAttribute::popFrom) for authentication? Or the old implementation should be suffice.

@Gaganjuneja chatted with Craig (@cwperks) offline, it seems like we may try to pursue a different path for early termination of the the request, I would suggest we do nothing for channel attributes (NettyAttribute::*) now since it is likely to go away (sadly no chances to fix that `2.11`). It should not be an issue, tracing is an experimental feature that is expected to have hiccups.

Reply:
@Gaganjuneja I've updated the change with respect to core changes